nat/traversal/pinger.go

```diff
@@ -104,43 +130,34 @@ func (p *Pinger) Start() {
 				}
 			}()
 
-			select {
-			case <-p.stop:
-				return
-			case <-time.After(pingInterval * time.Millisecond):
-			}
-
 			err = p.pingReceiver(conn)
 			if err != nil {
 				log.Error(prefix, "ping receiver error: ", err)
 				continue
 			}
 
-			p.pingReceived <- struct{}{}
 			log.Info(prefix, "ping received, waiting for a new connection")
-			conn.Close()
+
+			go p.natProxy.handOff(serviceType, conn)
 		}
 	}
 }
 
-// Stop stops the nat pinger
+// Stop noop method
 func (p *Pinger) Stop() {
-	p.once.Do(func() { close(p.stop) })
+	// noop method - NATPinger should not stop
```

Comment:
This is bad, you're leaving an infinite for loop hanging in some goroutine for no reason. We should stop gracefully.